pkg/obs/driver.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package obs
+
+import (
+	"fmt"
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"github.com/huaweicloud/huaweicloud-csi-driver/pkg/config"
+	"github.com/huaweicloud/huaweicloud-csi-driver/pkg/utils/metadatas"
+	"github.com/huaweicloud/huaweicloud-csi-driver/pkg/utils/mounts"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"k8s.io/klog/v2"
+)
+
+const (
+	driverName  = "obs.csi.huaweicloud.com"
+	topologyKey = "topology." + driverName + "/zone"
+	version     = "1.0.0"
+)
+
+type Driver struct {
+	name     string
+	nodeID   string
+	version  string
+	endpoint string
+	cloud    *config.CloudCredentials
+
+	ids *identityServer
+	cs  *controllerServer
+	ns  *nodeServer
+
+	vcap  []*csi.VolumeCapability_AccessMode
+	cscap []*csi.ControllerServiceCapability
+	nscap []*csi.NodeServiceCapability
+}
+
+func NewDriver(nodeID, endpoint string, cloud *config.CloudCredentials) *Driver {
+	klog.Infof("Driver: %v version: %v", driverName, version)
+
+	d := &Driver{}
+	d.name = driverName
+	d.nodeID = nodeID
+	d.version = version
+	d.endpoint = endpoint
+	d.cloud = cloud
+
+	d.AddControllerServiceCapabilities(
+		[]csi.ControllerServiceCapability_RPC_Type{
+			csi.ControllerServiceCapability_RPC_CREATE_DELETE_VOLUME,
+			csi.ControllerServiceCapability_RPC_LIST_VOLUMES,
+			csi.ControllerServiceCapability_RPC_GET_VOLUME,
+		})
+	d.AddVolumeCapabilityAccessModes([]csi.VolumeCapability_AccessMode_Mode{
```

Comment:
pls append VolumeCapability_AccessMode_MULTI_NODE_MULTI_WRITER and VolumeCapability_AccessMode_SINGLE_NODE_MULTI_WRITER

Reply:
done